### PR TITLE
PLT-881 Adds Legal and Support Settings to the System Console

### PR DESCRIPTION
**Pull request by rgarmsen2295**
Makes the following options configurable in the System Console in a new tab under a new tab called 'Legal and Support Settings', in addition to adding respective fields to the config file.

+ Terms of Service Link
+ Privacy Policy Link
+ About Link
+ Help Link
+ Report a Problem Link
+ Support Email

This also implements the functionality described in the help text for each of the above settings to a basic level.

When reviewing please double-check I've done the necessary back-end work for adding new fields to the config file, it's been a while since I've done so and the process has changed a bit!

PM Documentation:  To change the above default links/email to something more appropriate, a Mattermost System Admin can simply go to the System Console -> Settings -> Legal and Support Settings to find fields for each of the above options.  Upon saving their changes the links are then updated across all teams on that server (or removed if left blank).

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -82,6 +82,14 @@
         "ShowEmailAddress": true,
         "ShowFullName": true
     },
+    "LegalAndSupportSettings": {
+        "TermsOfServiceLink": "https://github.com/mattermost/platform/blob/master/README.md",
```
Comment (coreyhulen) [inline config/config.json]: these should start off linked to the static files

File: model/config.go [hunk 2]
```diff
@@ -113,6 +113,15 @@ type PrivacySettings struct {
 	ShowFullName     bool
 }
 
+type LegalAndSupportSettings struct {
+	TermsOfServiceLink string
+	PrivacyPolicyLink  string
+	AboutLink          string
+	HelpLink           string
+	ReportAProblemLink string
+	SupportEmail       string
```
Comment (coreyhulen) [inline model/config.go]: Is this different than feedback email?

File: docker/dev/config_docker.json [hunk 3]
```diff
@@ -82,6 +82,14 @@
         "ShowEmailAddress": true,
         "ShowFullName": true
     },
+    "LegalAndSupportSettings": {
```
Comment (coreyhulen) [inline docker/dev/config_docker.json]: This doesn't have to be a 1:1 mapping to the UI.  If it's just links then maybe LinkSettings (0/5)

File: docker/local/config_docker.json [hunk 4]
```diff
@@ -82,6 +82,14 @@
         "ShowEmailAddress": true,
         "ShowFullName": true
     },
+    "LegalAndSupportSettings": {
+        "TermsOfServiceLink": "https://github.com/mattermost/platform/blob/master/README.md",
+        "PrivacyPolicyLink": "https://github.com/mattermost/platform/blob/master/README.md",
+        "AboutLink": "http://www.mattermost.org/features/",
+        "HelpLink": "https://github.com/mattermost/platform/blob/master/doc/help/README.md",
+        "ReportAProblemLink": "https://forum.mattermost.org/c/general/trouble-shoot",
+        "SupportEmail": "feedback@mattermost.com"
```
Comment (coreyhulen) [inline docker/local/config_docker.json]: Not sure we want to be the first line of support.  I wonder if this should be empty to start.

File: config/config.json (hunk 1 above)
Comment (rgarmsen2295) [inline config/config.json]: The spec [here](https://mattermost.atlassian.net/browse/PLT-881) asked for these to be used as the default (for the time being).  Not against using the static files instead but we might want to have a discussion about the defaults then.

File: model/config.go (hunk 2 above)
Comment (rgarmsen2295) [inline model/config.go]: The spec/associated help text specifically mentions it only changing the email given during the tutorial.  It makes sense if we want to consolidate it down to one email, but if we want them to be separate (e.g. support handles emails sent from users needing help, whereas feedback is the email things like auto-generated notification emails are sent from) we will need to keep two separate settings.  Once again fine either way, but should probably have a discussion.

File: docker/dev/config_docker.json (hunk 3 above)
Comment (rgarmsen2295) [inline docker/dev/config_docker.json]: Same 0/5, only thoughts towards keeping it as is, is if we keep the `SupportEmail` field in there, or if we plan to put other settings there non-link related in the future.

File: docker/local/config_docker.json (hunk 4 above)
Comment (rgarmsen2295) [inline docker/local/config_docker.json]: I think IT posted about this on Core but it seems like another thing we should discuss then if we're not all on the same page about it.

**Comment by coreyhulen** [comment]
These should also be *string to detect when empty and set the defaults properly.

**Comment by rgarmsen2295** [comment]
Some of the help text specifically mentions that leaving it blank should cause the link to be removed from the UI, if that's what you mean (we might want to have a "Reset to Default" button in the sys console UI though too...)

**Comment by rgarmsen2295** [comment]
Update:  Going to use the previous static links by default and a PM will add documentation detailing how to change the links in the future from information in the PR description.  Will also rename the config file section to simply `SupportSettings` for better extensibility in the future.

**Comment by rgarmsen2295** [comment]
@coreyhulen Okay changes have been made, please take a look when you get a chance (especially at the default config stuff, tested it multiple times/ways but want to make sure I did it the right way).

**Comment by coreyhulen** [comment]
+1 looks good.  @lfbrock this has config file changes

**Comment by jwilander** [comment]
+1

**Pull request merged**